source/blog/2017-12-02-installing-etcd-as-a-system-container.md

```diff
@@ -0,0 +1,235 @@
+
+
+## Configuration at Install Time
+
+System Containers allow for options to be set at install time. These options may be used for templating file
+contents, file names, and directory names. How does one figure out what options are available? The
+``atomic`` command of course!
+
+```
+# atomic info --storage ostree registry.fedoraproject.org/f26/etcd
+Image Name: registry.fedoraproject.org/f26/etcd:latest
+INSTALL: /usr/bin/docker run --rm $OPT1 --privileged -v /:/host -e HOST=/host -e NAME=$NAME -e IMAGE=$IMAGE $IMAGE $OPT2 /usr/bin/install.sh  $OPT3
+RUN: /usr/bin/docker run -d $OPT1 -p 4001:4001 -p 7001:7001 -p 2379:2379 -p 2380:2380 --name $NAME $IMAGE $OPT2 $OPT3
+UNINSTALL: /usr/bin/docker run --rm $OPT1 --privileged -v /:/host -e HOST=/host -e NAME=$NAME -e IMAGE=$IMAGE $IMAGE $OPT2 /usr/bin/uninstall.sh $OPT3
+architecture: x86_64
+authoritative-source-url: registry.example.com
+build-date: 2017-09-21T19:01:39.071232
+com.redhat.build-host: osbs-node01.phx2.fedoraproject.org
+com.redhat.component: etcd
+distribution-scope: private
+install: /usr/bin/docker run --rm \$OPT1 --privileged -v /:/host -e HOST=/host -e NAME=\$NAME -e IMAGE=\$IMAGE \$IMAGE \$OPT2 /usr/bin/install.sh  \$OPT3
+license: MIT
+maintainer: Giuseppe Scrivano <gscrivan@redhat.com>
+name: f26/etcd
+release: 8.f26container
+run: /usr/bin/docker run -d \$OPT1 -p 4001:4001 -p 7001:7001 -p 2379:2379 -p 2380:2380 --name \$NAME \$IMAGE \$OPT2 \$OPT3
+summary: A key-value store for shared configuration and service discovery.
+uninstall: /usr/bin/docker run --rm \$OPT1 --privileged -v /:/host -e HOST=/host -e NAME=\$NAME -e IMAGE=\$IMAGE \$IMAGE \$OPT2 /usr/bin/uninstall.sh \$OPT3
+vcs-ref: 51e6ea6b1aca965dbc6bd6c1b3e15bb271f5c99f
+vcs-type: git
+vendor: Fedora Project
+version: 0
+
+
+Template variables with default value, but overridable with --set:
+ADDTL_MOUNTS: 
+CONF_DIRECTORY: {SET_BY_OS}
+ETCD_ADVERTISE_CLIENT_URLS: 
+ETCD_CERT_FILE: 
+ETCD_CLIENT_CERT_AUTH: false
+ETCD_CORS: 
+ETCD_DATA_DIR: 
+ETCD_DEBUG: false
+ETCD_DISCOVERY: 
+ETCD_DISCOVERY_FALLBACK: proxy
+ETCD_DISCOVERY_PROXY: 
+ETCD_DISCOVERY_SRV: 
+ETCD_ELECTION_TIMEOUT: 1000
+ETCD_ENABLE_PPROF: false
+ETCD_HEARTBEAT_INTERVAL: 100
+ETCD_INITIAL_ADVERTISE_PEER_URLS: 
+ETCD_INITIAL_CLUSTER: 
+ETCD_INITIAL_CLUSTER_STATE: new
+ETCD_INITIAL_CLUSTER_TOKEN: etcd-cluster
+ETCD_KEY_FILE: 
+ETCD_LISTEN_CLIENT_URLS: 
+ETCD_LISTEN_PEER_URLS: 
+ETCD_LOG_PACKAGE_LEVELS: 
+ETCD_MAX_SNAPSHOTS: 5
+ETCD_MAX_WALS: 5
+ETCD_NAME: 
+ETCD_PEER_CERT_FILE: 
+ETCD_PEER_CLIENT_CERT_AUTH: false
+ETCD_PEER_KEY_FILE: 
+ETCD_PEER_TRUSTED_CA_FILE: 
+ETCD_PROXY: off
+ETCD_PROXY_DIAL_TIMEOUT: 1000
+ETCD_PROXY_FAILURE_WAIT: 5000
+ETCD_PROXY_READ_TIMEOUT: 0
+ETCD_PROXY_REFRESH_INTERVAL: 30000
+ETCD_PROXY_WRITE_TIMEOUT: 5000
+ETCD_SNAPSHOT_COUNT: 10000
+ETCD_STRICT_RECONFIG_CHECK: false
+ETCD_TRUSTED_CA_FILE: 
+ETCD_WAL_DIR: 
+PIDFILE: {SET_BY_OS}
+RUN_DIRECTORY: {SET_BY_OS}
+STATE_DIRECTORY: {SET_BY_OS}
+UUID: {SET_BY_OS}
+```
```

Comment:
This seems like a lot of terminal text for a blog post.  Can we truncate/compress it down to the relevant elements, with ... to indicate the parts we've left out?

Reply:
Sure

Reply:
Done